qt/applications/workbench/workbench/plugins/workspacewidget.py

```diff
@@ -121,10 +123,19 @@ def _do_show_instrument(self, names):
 
     def _do_show_data(self, names):
         for ws in self._ads.retrieveWorkspaces(names, unrollGroups=True):
-            # the plot function is being injected in the presenter
-            # this is done so that the plotting library is mockable in testing
-            presenter = MatrixWorkspaceDisplay(ws, plot=plot, parent=self)
-            presenter.view.show()
+            try:
+                MatrixWorkspaceDisplay.supports(ws)
+                # the plot function is being injected in the presenter
+                # this is done so that the plotting library is mockable in testing
+                presenter = MatrixWorkspaceDisplay(ws, plot=plot, parent=self)
+                presenter.view.show()
+            except ValueError:
+                try:
+                    TableWorkspaceDisplay.supports(ws)
+                    presenter = TableWorkspaceDisplay(ws, plot=matplotlib.pyplot, parent=self)
```

Comment:
Does this need to use `matplotlib.pyplot` directly? Can it use `plot` like the `MatrixWorkspaceDisplay`? It looks a little odd at first glance that they use different things.

Reply:
`pyplot` is used directly to handle the plotting logic manually, a lot of the code in `plotfunctions` assumes it is working with MatrixWorkspaces. 

I am not sure how hard it would be to add support for TableWorkspaces, but the implementation will branch out from MatrixWorkspaces significantly. I think it is only worth it if there is an external way of plotting TableWorkspaces, like right clicking in the WorkspaceWidget (which is not currently available in either MantidPlot or Workbench)